lib/rack/request.rb

```diff
@@ -47,59 +43,71 @@ def delete_param(k)
       v
     end
 
-    # Predicate method to test to see if `name` has been set as request
-    # specific data
-    def has_header?(name)
-      @env.key? name
-    end
+    module Env
+      # The environment of the request.
+      attr_reader :env
 
-    # Get a request specific value for `name`.
-    def get_header(name)
-      @env[name]
-    end
+      def initialize(env)
```

Comment:
If this is supposed to be a mixin, why do we even define `initialize`?

Surely this is up to the consumer to set `@env`.

Reply:
Agreed that it's not what I expected when I entered the code base. I think things organically evolved as opposed to a coordinated conversation around how Rack's core is intended to be used/extended and with which patterns and assumptions.

This isn't my show so I'd be interested to hear how Rack's core team would intend/prefer the components of it to be extended. Ruby offers some powerful but complex inheritance and mixing tools that are fun to use, but I always prefer explicit encapsulation as I tend to not remember fancier programming after 3 months of absence.

Reply:
Does it break Rails if we remove it?